### PR TITLE
40 pull parser

### PR DESCRIPTION
**Pull request by kordano**

### CODE REVIEW

File: src/datahike/pull_api.cljc [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-    [datahike.db :as db]
-    [datahike.datom :as dd]
-    [datahike.pull-parser :as dpp #?@(:cljs [:refer [PullSpec]])])
-    #?(:clj
-      (:import
-        [datahike.datom Datom]
-        [datahike.pull_parser PullSpec])))
+   [datahike.db :as db]
+   [datahike.datom :as dd]
+   #?@(:cljs [datalog.parse.type :refer [PullSpec]])
```
Comment (grischoun) [inline src/datahike/pull_api.cljc]: Isn't this supposed to be `.parser.`?
And also `PullSpec` does not exist in `datalog.parser.type`.

File: src/datahike/pull_api.cljc [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+   [datalog.parser.pull :as dpp])
```
Comment (grischoun) [inline src/datahike/pull_api.cljc]: The alias `dpp` is never used later in the file.

File: src/datahike/pull_api.cljc [hunk 3]
```diff
@@ -272,8 +273,8 @@
 
 (defn pull [db selector eid]
   {:pre [(db/db? db)]}
-  (pull-spec db (dpp/parse-pull selector) [eid] false))
+  (pull-spec db (datalog.parser.pull/parse-pull selector)  [eid] false))
```
Comment (grischoun) [inline src/datahike/pull_api.cljc]: use `dpp` instead?

File: src/datahike/pull_api.cljc [hunk 4]
```diff
@@ -272,8 +273,8 @@
...
 
 (defn pull-many [db selector eids]
   {:pre [(db/db? db)]}
-  (pull-spec db (dpp/parse-pull selector) eids true))
+  (pull-spec db (datalog.parser.pull/parse-pull selector) eids true))
```
Comment (grischoun) [inline src/datahike/pull_api.cljc]: use `dpp` instead?

File: src/datahike/pull_api.cljc (hunk 1 above)
Comment (grischoun) [reply src/datahike/pull_api.cljc]: Also, is `type` the right name here?

**Comment by kordano** [comment]
thanks for the feedback. I will fix the issues.

File: src/datahike/pull_api.cljc (hunk 1 above)
Comment (kordano) [reply src/datahike/pull_api.cljc]: You're correct, it is the wrong namespace.

File: src/datahike/pull_api.cljc (hunk 2 above)
Comment (kordano) [reply src/datahike/pull_api.cljc]: I adjusted that.

**Pull request merged**